### ISSUE TITLE
[Bug]: Shows with Same Name Don't All Appear in Manage Fingerprints

### ISSUE THREAD

**Issue by ethaldeman** [intro-skipper/intro-skipper#179]
### Describe the bug

I have the original Cowboy Bebop anime and the Netflix live action remake in my Jellyfin library. In Jellyfin both shows are unique series, but are both known as "Cowboy Bebop". They are stored locally in directories with the release year in the name. Both have intro data and I have skipped the intro on both using the "Skip Intro" button. However under Manage Fingerprints, I only see one listing for "Cowboy Bebop", and it appears to contain data for the anime. I have no way to edit finger prints for the live action version. 

### Jellyfin installation method

Ubuntu

### Container image and tag

jellyfin 10.9.2

### Operating System

 Ubuntu 22.04.4

### IMDb ID of that TV Series

tt0213338 and tt1267295

### Support Bundle

```shell
* Jellyfin version: 10.9.2
* Plugin version: 0.2.0+46f280f0afae
* Queue contents: 4918 episodes, 266 seasons
* Warnings: `None`
* FFmpeg: `okay`

FFmpeg version:

ffmpeg version 6.0.1-Jellyfin Copyright (c) 2000-2023 the FFmpeg developers
built with gcc 11 (Ubuntu 11.4.0-1ubuntu1~22.04)
configuration: --prefix=/usr/lib/jellyfin-ffmpeg --target-os=linux --extra-version=Jellyfin --disable-doc --disable-ffplay --disable-ptx-compression --disable-static --disable-libxcb --disable-sdl2 --disable-xlib --enable-lto --enable-gpl --enable-version3 --enable-shared --enable-gmp --enable-gnutls --enable-chromaprint --enable-opencl --enable-libdrm --enable-libass --enable-libfreetype --enable-libfribidi --enable-libfontconfig --enable-libbluray --enable-libmp3lame --enable-libopus --enable-libtheora --enable-libvorbis --enable-libopenmpt --enable-libdav1d --enable-libsvtav1 --enable-libwebp --enable-libvpx --enable-libx264 --enable-libx265 --enable-libzvbi --enable-libzimg --enable-libfdk-aac --arch=amd64 --enable-libshaderc --enable-libplacebo --enable-vulkan --enable-vaapi --enable-amf --enable-libvpl --enable-ffnvcodec --enable-cuda --enable-cuda-llvm --enable-cuvid --enable-nvdec --enable-nvenc
libavutil      58.  2.100 / 58.  2.100
libavcodec     60.  3.100 / 60.  3.100
libavformat    60.  3.100 / 60.  3.100
libavdevice    60.  1.100 / 60.  1.100
libavfilter     9.  3.100 /  9.  3.100
libswscale      7.  1.100 /  7.  1.100
libswresample   4. 10.100 /  4. 10.100
libpostproc    57.  1.100 / 57.  1.100
```
```


### Jellyfin logs

_No response_

**Comment by AbandonedCart** [comment]
I have a feeling this is the fontend portion causing the issue because behind the scenes, everything is given an ID. Is it putting all of the seasons from both into the one or only displaying the one and nothing else?

**Comment by AbandonedCart** [comment]
I'm still trying to figure out why exactly it wouldn't work. Based on the code, it should at least end up throwing everything under one heading because it adds the shows, not put or set (in code terms, that means it's not targeting the name, but simply throwing it on the list). If anything, you should be ending up wits a duplicate.

**Comment by jumoog** [comment]
Okay, so they both have the same name, `Cowboy Bebop`, but you have 2 different TV series in your library?

**Comment by AbandonedCart** [comment]
(Yes)

**Comment by ethaldeman** [comment]
> Okay, so they both have the same name, `Cowboy Bebop`, but you have 2 different TV series in your library?

Yes, below are the two series in Jellyfin. 

![image](https://github.com/jumoog/intro-skipper/assets/31450528/d694e735-45f5-4538-a4c9-cf4c834b88b2)

Below is the the single listing in the finger print editor.

![image](https://github.com/jumoog/intro-skipper/assets/31450528/b5dcffbc-5ef9-42ff-abae-b7f112086348)

**Comment by jumoog** [comment]
the key is identical and is therefore overwritten again. May be we have to add the year to the key.

https://github.com/jumoog/intro-skipper/blob/9974e840bbe6d82799d70fa6364b84ba84940796/ConfusedPolarBear.Plugin.IntroSkipper/Controllers/VisualizationController.cs#L68

**Comment by jumoog** [comment]
> > Okay, so they both have the same name, `Cowboy Bebop`, but you have 2 different TV series in your library?
> 
> Yes, below are the two series in Jellyfin.
> 
> ![image](https://private-user-images.githubusercontent.com/31450528/333909617-d694e735-45f5-4538-a4c9-cf4c834b88b2.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTY4MDU0NzUsIm5iZiI6MTcxNjgwNTE3NSwicGF0aCI6Ii8zMTQ1MDUyOC8zMzM5MDk2MTctZDY5NGU3MzUtNDVmNS00NTM4LWE0YzktY2Y0YzgzNGI4OGIyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MjclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTI3VDEwMTkzNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWEyODYxMWMzZmU4MDgyMDMxMmI5NjMzZGJlNjZjZGY1OTI4ZDEzMWViNGU0YmM0MmViMjAzNmU1MzdlOGNkMmImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.XRaRTpgSPJ77aiXTstWKcgV1kya9pZDfp10I9IpaROA)
> 
> Below is the the single listing in the finger print editor.
> 
> ![image](https://private-user-images.githubusercontent.com/31450528/333909669-b5dcffbc-5ef9-42ff-abae-b7f112086348.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTY4MDU0NzUsIm5iZiI6MTcxNjgwNTE3NSwicGF0aCI6Ii8zMTQ1MDUyOC8zMzM5MDk2NjktYjVkY2ZmYmMtNWVmOS00MmZmLWFiYWUtYjdmMTEyMDg2MzQ4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MjclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTI3VDEwMTkzNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTExNTJkMjFhMzBjYjk0Y2ViODJiOThkNTQ3N2M3NDgyMWNlYTRjOTI5Njk4ZjkwYmFjYjU2ZDU3NDc4ZWFhYjcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.aUI0bK9l9ZSOdmYgHQxoO-dMEHD4gVHRCRc56MCQLBQ)

As a workaround, you can rename the one TV series to `Cowboy Bebop (2021)` in the Metadata.

**Comment by AbandonedCart** [comment]
We may need to add key value pairs to everything with how many shows Netflix adapts. One piece is another one it will trip over.

**Comment by jumoog** [comment]
Some shows I have as 1080p and 4K. To do it right, we would even have to separate the library.

**Comment by AbandonedCart** [comment]
But that also means it's not really something you have to handle. Saying that it's not compatible with multiple resolutions that have different lengths makes sense. Saying you can't have two series that have the same name doesn't.

The problem is being stuck between the single reference preventing unintentional duplicates, but at the same time also preventing intentional ones.

**Comment by AbandonedCart** [comment]
Isn't the issue that one Cowboy Bebop doesn't come up at all? Everything works fine, except the editor.

Seems like the simple solution would be to assign the series ID to the key alongside the name.

**Comment by AbandonedCart** [comment]
Yes, so that has nothing to do with multiple resolutions. The issue is that the key is the series name and there are two that are identical.

**Comment by jumoog** [comment]
> Yes, so that has nothing to do with multiple resolutions. The issue is that the key is the series name and there are two that are identical.

It is similar. I also have two identical series names.

**Comment by AbandonedCart** [comment]
I'm not sure where you are going with all of this, but the solution seems to be the same.

**Comment by AbandonedCart** [comment]
There are a few minor roadblocks. 

By default, folders aren't really given any sort of universally unique identification beyond the path. The identification they get as part of Jellyfin is based on the scrapers used, which could vary user to user and are no more reliable than the final name they decide. This means that folder 1 and folder 2 are technically indistinguishable in code as anything more than folder 1 and folder 2 without intervention.

That leaves us at the only real way to separate the two when collecting names would be to also assign the path alongside it. That just seems like it will end up in disaster since the premise of libraries is that the same item with a single path can exist in multiple libraries. The time should be the same for all of them, sure, but moving a file collapses the whole thing.

It seems like the most rational solution may be the original workaround. When two series have the same name, something like (2019) in the folder name would create a clear separation that doesn't rely on every identification being perfect.

**Comment by ethaldeman** [comment]
> TL;DR I think if we try to address it on this end, it will end up being a source of problems.

So does that mean you won't be fixing the problem within the addon, and rely on the end-user to uniquely name their series within Jellyfin?

> When two series have the same name, something like (2019) in the folder name would create a clear separation that doesn't rely on every identification being perfect.

Or will you implement the above logic within the addon in some future update?

**Comment by AbandonedCart** [comment]
The logic you quoted was actually referring to user intervention. The irony of the situation is almost any solution the plugin could offer relies on there first being some action taken or at least verified. At that point, it's not really any more automated.
